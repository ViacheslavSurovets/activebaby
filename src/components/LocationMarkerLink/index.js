export { default as LocationMarkerLink } from './LocationMarkerLink';
