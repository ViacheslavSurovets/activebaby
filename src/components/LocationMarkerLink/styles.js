import styled from 'styled-components';

export const MapIconWrapper = styled.figure`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  width:100%;
  
  
  &:hover {
    svg path {
      fill: ${ ( { theme } ) => theme.header.colorOnHover }
    }
  }
`;