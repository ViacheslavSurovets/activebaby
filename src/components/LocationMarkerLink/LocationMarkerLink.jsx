import React from 'react';
import { LocationMarkerIcon } from '@components';
import { withRouter } from 'react-router-dom';
import { MapIconWrapper } from './styles';
import PropTypes from 'prop-types';


const LocationMarkerLink = ( { history } ) => (
  <MapIconWrapper onClick={ () => history.push ( '/map' ) }>
    <LocationMarkerIcon />
  </MapIconWrapper>
);

export default withRouter ( LocationMarkerLink );

LocationMarkerLink.propTypes = {
  history: PropTypes.object
};
