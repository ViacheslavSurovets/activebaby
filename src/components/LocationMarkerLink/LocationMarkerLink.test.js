import React from 'react';
import { shallow } from 'enzyme';
import LocationMarkerLink from './LocationMarkerLink';


describe ( 'LocationMarkerLink', () => {
  it ( 'LocationMarkerLink Snapshot', () => {
    expect ( shallow ( <LocationMarkerLink /> ) ).toMatchSnapshot ();
  } );
} );
