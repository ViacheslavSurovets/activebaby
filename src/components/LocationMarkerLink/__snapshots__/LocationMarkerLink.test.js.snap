// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationMarkerLink LocationMarkerLink Snapshot 1`] = `ShallowWrapper {}`;
